### ISSUE TITLE
🤾‍♀️ [Feature] Most interop methods should be marked with `catch`

### ISSUE THREAD

**Issue by Jasper-Bekkers** [cloudflare/workers-rs#545]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Given how prevalent it is in the node.js side of things to handle errors through exceptions, I think we should mark most methods in the worker-sys crate as `catch`. From the documentation, emphasis mine.

> By default wasm-bindgen **will take no action** when wasm calls a JS function which ends up throwing an exception. The wasm spec right now doesn't support stack unwinding and as a result Rust code will not execute destructors. This can unfortunately cause memory leaks in Rust right now, but as soon as wasm implements catching exceptions we'll be sure to add support as well!

https://rustwasm.github.io/wasm-bindgen/reference/attributes/on-js-imports/catch.html

Not handling catching these exceptions leads to the (quite frustrating) error message when developing.

> A hanging Promise was canceled. This happens when the worker runtime is waiting for a Promise from JavaScript to resolve, but has detected that the Promise cannot possibly ever resolve because all code and events related to the Promise's I/O context have already finished.                      
> X [ERROR] Uncaught (in response) Error: The script will never generate a response.                    

Obviously the error message is somewhat correct: the wasm module terminated without any reason as to why. Diving deeper and doing some debugging I found that this is mostly just because I'm passing bad inputs into the api's which are then detected by the node.js code, and bubbled up to us but we silently ignore them.  

However, getting `A hanging Promise was canceled.` messages because - say - my `part_number` is 0 when it should be 1 isn't helpful.

This would be quite a sweeping change though, and most likely cause a lot of `.unwrap` or `.expect` calls to have to be added to the wrapping library, but it would make development quite a bit nicer for the folks using this library so I'd like to discuss possible ways forward. I don't think `wasm-bindgen` has a catch-all way of dealing with this unfortunately (not that I could find at least).

**Comment by Jasper-Bekkers** [comment]
Some stats: currently there are 166 methods exposed through `worker-sys` and 39 of those have the `catch` attribute.

**Comment by Jasper-Bekkers** [comment]
https://github.com/cloudflare/workers-rs/pull/546 Is an initial attempt at trying to fix these issues.

**Comment by Jasper-Bekkers** [comment]
Example from my follow up #547. With these changes if you pass an object with `part_number` into `complete` you'll get this error message:

```
Error: TypeError: Part number must be between 1 and 10000 (inclusive). Actual value was: 0 - Cause: TypeError: Part number must be between 1 and 10000 (inclusive). Actual value was: 0
```

Whereas if you'd rebase that PR on latest main you'd just get

```
A hanging Promise was canceled. This happens when the worker runtime is waiting for a Promise from JavaScript to resolve, but has detected that the Promise cannot possibly ever resolve because all code and events related to the Promise's I/O context have already finished.
```

**Comment by Jasper-Bekkers** [comment]
Closed by #546
